### ISSUE TITLE
Add scripts to run `flutter test` and `flutter analyze` on all files before pushing.

### ISSUE THREAD

**Issue by AliAkberAakash** [getsentry/sentry-dart#822]
This feature is for repository maintenance.

Before pushing I run `flutter test` but often forget to run `flutter analyze`. Also doing this manually doing `cd` in each folder is a bit cumbersome. It would be a lot easier to just run a script and do this for all files. This would also decrease frequent pushes to PR's .

Also if there is a plan to do this, can I be assigned the task?

**Comment by marandaneto** [comment]
There is already, do you mean this?
https://github.com/getsentry/sentry-dart/blob/main/dart/tool/presubmit.sh

**Comment by AliAkberAakash** [comment]
This is for the dart module only? Or does this run for every module?

**Comment by marandaneto** [comment]
> This is for the dart module only? Or does this run for every module?

Indeed, it's not executed once and works for all modules, that could be improved.
You'd need to do it for each folder.
Another option is having a GH Action that executes and commits missing formats in the code on CI, so you don't need to bother about doing it locally, e.g. https://github.com/getsentry/sentry-java/pull/1963/files

**Comment by buenaflor** [comment]
Closing as we already use pre-commit hooks with melos
